### ISSUE TITLE
Error: Unexpected end of JSON input

### ISSUE THREAD

**Issue by kristoferbaxter** [ampproject/amphtml#20690]
https://go/ampe/CK6EmuuQ5vKB3gE

Before attempting to call `Response.json()` validate the `Response` object contains a body.

```
Error: Unexpected end of JSON input
    at unexpected-end-of-json-input.js:1:1
```

Impacted code block
```javascript
function nb(a, b, c, d) {
            var e = ".";
            e = void 0 === e ? "." : e;
            c = void 0 === c ? 0 : c;
            d = void 0 === d ? !1 : d;
            eb(b.getAttribute("src"), b);
            var f = I(a.win, "batched-xhr");
            return ob(b, c, d).then(function(a) {
                return f.fetchJson(a.xhrUrl, a.fetchOpt)
            }).then(function(a) {
                return a.json()
            }).then(function(a) {
                if (null == a)
                    throw Error("Response is undefined.");
                return mb(a, e || ".")
            })
        }
```

**Comment by ampprojectbot** [comment]
This issue doesn't have a category which makes it harder for us to keep track of it. @prateekbh Please add an appropriate category.

**Comment by jridgewell** [comment]
Reassigning to @cathyxz 

This is coming from https://github.com/ampproject/amphtml/blob/93e0eb841e1eafbd5ab6b6bc45eeddc98608622b/extensions/amp-list/0.1/amp-list.js#L465-L467, which was introduced in #19770.

We should brand all fetch errors as user events, including this `res.json()`.

**Comment by jridgewell** [comment]
@cathyxz: https://nv-ua.cdn.ampproject.org/c/s/nv.ua/ukr/amp/eks-sotrudnik-razvedki-ssha-priznalsya-v-popytke-shpionazha-v-polzu-kitaya-50011167.html?next-page reproduces this easily, just scroll down the page
